File: services/palace-mcp/tests/extractors/fixtures/ts-mini-project/src/greeter.ts
export interface Greeting {
  message: string;
  recipient: string;
}

export class Greeter {
  private prefix: string;

  constructor(prefix: string) {
    this.prefix = prefix;
  }

  greet(name: string): Greeting {
    return {
      message: `${this.prefix} ${name}`,
      recipient: name,
    };
  }

  greetAll(names: string[]): Greeting[] {
    return names.map((n) => this.greet(n));
  }
}

export function formatGreeting(g: Greeting): string {
  return `[${g.recipient}] ${g.message}`;
}

File: services/palace-mcp/tests/extractors/fixtures/py-mini-project/src/pymini/greeter.py
"""Greeter module — mirrors ts-mini-project/src/greeter.ts structure."""

from dataclasses import dataclass


@dataclass
class Greeting:
    message: str
    recipient: str


class Greeter:
    def __init__(self, prefix: str) -> None:
        self.prefix = prefix

    def greet(self, name: str) -> Greeting:
        return Greeting(message=f"{self.prefix} {name}", recipient=name)

    def greet_all(self, names: list[str]) -> list[Greeting]:
        return [self.greet(n) for n in names]


def format_greeting(g: Greeting) -> str:
    return f"[{g.recipient}] {g.message}"
